subnet/etcd.go

```diff
@@ -347,36 +390,118 @@ func parseSubnetWatchResponse(resp *etcd.Response) (WatchResult, error) {
 		}
 
 		evt = Event{
-			SubnetAdded,
+			EventAdded,
 			Lease{
 				Subnet:     sn,
 				Attrs:      attrs,
 				Expiration: exp,
 			},
+			"",
 		}
 	}
 
-	cursor := watchCursor{resp.Node.ModifiedIndex}
+	return LeaseWatchResult{
+		Cursor: watchCursor{resp.Node.ModifiedIndex},
+		Events: []Event{evt},
+	}, nil
+}
+
+// Returns network name from the 'config' key
+func (m *EtcdManager) parseNetworkKey(s string) (string, error) {
+	if parts := m.networkRegex.FindStringSubmatch(s); len(parts) == 2 {
+		return parts[1], nil
+	}
+
+	return "", errors.New("Error parsing Network key")
+}
+
+func (m *EtcdManager) parseNetworkWatchResponse(resp *etcd.Response) (NetworkWatchResult, error) {
+	netname, err := m.parseNetworkKey(resp.Node.Key)
```

Comment:
The watch is recursive on something like `/coreos.com/network/`. Wouldn't then this fire on every key change, like subnets getting added? In that case `ParseConfig` will fail and that error propagated.